servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -167,8 +169,11 @@ final class DefaultLoadBalancer<ResolvedAddress, C extends LoadBalancedConnectio
         subscribeToEvents(false);
 
         // When we get a health-status event we should update the host set.
-        this.outlierDetector.healthStatusChanged().forEach((ignored) ->
+        this.outlierDetectorStatusChangeStream = this.outlierDetector.healthStatusChanged().forEach((ignored) ->
             sequentialExecutor.execute(() -> sequentialUpdateUsedHosts(usedHosts)));
```

Comment:
Previously this was simply discarded which was fine because of the way it's implemented but we might as well cancel it in case that changes in the future.

Reply:
Good catch. Up to you if you'd like to keep it here or move into a separate PR for visibility.